src/Sentry.AspNetCore/SentryTracingMiddleware.cs

```diff
@@ -35,74 +38,102 @@ internal class SentryTracingMiddleware
                 return null;
             }
         }
 
-        /// <summary>
-        /// Handles the <see cref="HttpContext"/>.
-        /// </summary>
-        public async Task InvokeAsync(HttpContext context)
+        private ITransaction? TryStartTransaction(HttpContext context)
         {
-            var hub = _getHub();
-            if (!hub.IsEnabled)
+            try
             {
-                await _next(context).ConfigureAwait(false);
-                return;
-            }
+                var hub = _getHub();
 
-            await hub.ConfigureScopeAsync(async scope =>
-            {
                 // Attempt to start a transaction from the trace header if it exists
                 var traceHeader = TryGetSentryTraceHeader(context);
 
-                // Defer setting name until other middlewares have finished
+                // It's important to try and set the transaction name
+                // to some value here so that it's available for use
+                // in sampling.
+                // At a later stage, we will try to get the transaction name
+                // again, to account for the other middlewares that may have
+                // ran after ours.
+                var transactionName =
+                    context.TryGetTransactionName() ??
+                    UnknownRouteTransactionName;
+
                 var transaction = traceHeader is not null
                     ? hub.StartTransaction(
-                        "Unknown Route",
-                        "http.server",
+                        transactionName,
+                        OperationName,
                         traceHeader
                     )
                     : hub.StartTransaction(
-                        "Unknown Route",
-                        "http.server"
+                        transactionName,
+                        OperationName
                     );
 
                 _options.DiagnosticLogger?.LogInfo(
-                    "Started transaction: Span ID: {0}, Trace ID: {1}",
+                    "Started transaction with span ID '{0}' and trace ID '{1}'.",
                     transaction.SpanId,
                     transaction.TraceId
                 );
 
-                // Put the transaction on the scope
-                scope.Transaction = transaction;
+                return transaction;
+            }
+            catch (Exception ex)
+            {
+                _options.DiagnosticLogger?.LogError("Failed to start transaction.", ex);
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Handles the <see cref="HttpContext"/>.
+        /// </summary>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var hub = _getHub();
+
+            if (!hub.IsEnabled)
+            {
+                await _next(context).ConfigureAwait(false);
+                return;
+            }
+
+            using (hub.PushAndLockScope())
```

Comment:
isn't locking done by the error SDK? Do we want to do this here?

I think this whole locking was a bad idea. Better would have been to disconenct the MEL BeginScope with Sentry's PushScope

Reply:
Hm yeah. So should I use `ConfigureScopeAsync(...)` here instead? In the original issue, according to the logs, it failed for some reason. Maybe because the scope was locked?

Reply:
I think you don't need to push a scope at all, instead just use the current scope available?

Though this would depend on the first middleware running. So maybe we just go ahead like this